### DISCUSSION TITLE
Upload to AWS s3 bucket works but the app throws an error and doesn't display the image

### DISCUSSION THREAD

**Discussion by mbelang** [outline/outline#1591] [General]
When uploading and image to s3 bucket via the app using `/image` command, the image is sent to the bucket but the app throws an error saying that an error occurred and the image is never displayed.

**To Reproduce**
1. Go to any page
2. enter `/image` command
3. Chose an image
4. An error occurred uploading the image

**Expected behavior**
The uploaded image is displayed on the page

**Screenshots**
![outline](https://user-images.githubusercontent.com/5888932/95874682-26346200-0d3f-11eb-8418-2d112b7eb002.png)

**Outline (please complete the following information):**
- Install: self hosted on kubernetes (AWS EKS)
- Version: 0.48.1 (From DockerHub official outlinewiki image)

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Browser: Chromium
 - Version: 86.0.4240.75 (Official Build) Arch Linux (64-bit)

**Additional details**
I didn't find the way to activate DEBUG mode with the official docker image, It always fails to load outline for missing koa dev libraries which makes sens but there should really be a way to have log verbosity configuration for production environment.

**Comment by tommoor** [comment]
> I didn't find the way to activate DEBUG mode

Set `DEBUG=*` as an environment variable. But the frontend should have a JS/network error with enough details to gauge what's going on.

I move this to a discussion as it's not a bug with the software.

**Reply by mbelang** [reply]
`DEBUG=*` doesn't work. The server complains about missing koa dev libraries.

for the Frontend, I really forgot to check that.

`Access to fetch at 'https://s3.ca-central-1.amazonaws.com/my-outline-bucket' from origin 'https://outline.dev.example.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

Does the bucket needs to have `example.com` in the name in order to work?

**Reply by mbelang** [reply]
Or in fact, do I have to serve content of the bucket via AWS CloudFront?

**Reply by tommoor** [reply]
You don't need to serve via cloudfront, the CORS policy can be added to the bucket directly. You can start with the entire domain but could also restrict to the specific subdomain where the wiki is hosted – probably doesn't matter too much if you're in control of both.

**Reply by tommoor** [reply]
> The server complains about missing koa dev libraries.

I'm intrigued if you can provide these.

**Reply by mbelang** [reply]
Ok DEBUG=* works fine... I was using 

I was using this from the Readme: `DEBUG=sql,cache,presenters,events,logistics,emails,mailer` and that failed.

Still, how do I resolve the CORS issue? I follow this and it doesn't work https://github.com/outline/outline/pull/1137. Maybe I'm missing something for the configuration / env?

**Reply by tommoor** [reply]
This is a setting in AWS, not Outline.

https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html

**Reply by mbelang** [reply]
> You don't need to serve via cloudfront, the CORS policy can be added to the bucket directly. You can start with the entire domain but could also restrict to the specific subdomain where the wiki is hosted – probably doesn't matter too much if you're in control of both.

Sorry I totally missed that.

**Reply by mbelang** [reply]
```
<CORSConfiguration>
 <CORSRule>
   <AllowedOrigin>https://my-cool.domain.com</AllowedOrigin>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedMethod>HEAD</AllowedMethod>
   <AllowedMethod>POST</AllowedMethod>
 </CORSRule>
</CORSConfiguration>
```

The above fixed the problem. Thanks for your help.
